Fix bugs in Skew map

knelli2 opened the pull request:
## Proposed changes

Found these as I was testing #6487. Figured I'd factor them out.

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

knelli2 committed:
Add small optimization for skew map

knelli2 committed:
Change minus sign in Skew map

The control system expects it this way, not the previous way

knelli2 committed:
Fix bug in Skew inverse

The inverse needs to be allowed to evaluate the map beyond the
outer radius during the root find. Previously, it would error.

wthrowe commented on src/Domain/CoordinateMaps/TimeDependent/Skew.hpp:
```diff
@@ -213,8 +213,8 @@ class Skew {
-  tt::remove_cvref_wrap_t<T> get_width(
-      const std::array<T, 3>& source_coords) const;
+  tt::remove_cvref_wrap_t<T> get_width(const std::array<T, 3>& source_coords,
+                                       bool ignore_error = false) const;
```
This is in the wrong commit.

wthrowe commented on tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Skew.cpp:
```diff
@@ -73,16 +82,27 @@ void test(const gsl::not_null<Generator*> generator) {
+    }();
+
+    const std::array<DataVector, 3> dv_point_xi = [&]() {
+      {
```
[optional] Remove extra braces.

wthrowe commented on src/Domain/CoordinateMaps/TimeDependent/Skew.cpp:
```diff
@@ -80,8 +80,9 @@ std::optional<std::array<double, 3>> Skew::inverse(
   }
```
Just to make sure I'm understanding correctly: The sign being changed here is the sign of the FoT, correct?

knelli2 replied:
Yes it's the sign of the FoT which has changed, which propagates out the overall sign of the sum `tan(F_y) * (y-y_C) + tan(F_z) * (z-z_C)` because of the `tan`s

knelli2 committed:
Capture more vars in jacobian test helper

Also allow approx for inverse test helper

knelli2 committed:
Make singular skew map check more robust

This checks if the first deriv is negative (singular map) at the
point the second deriv is zero (does a root find to get this point).

knelli2 commented:
@wthrowe I squashed in your comments, and also added a couple new commits pertaining to a better check for a singular map.

wthrowe commented on src/Domain/CoordinateMaps/TimeDependent/Skew.cpp:
```diff
@@ -344,23 +346,93 @@ void Skew::check_for_singular_map(
+      return M_PI * one_over_outer_radius_squared_ * get_element(tan_sum, i) *
+             (sin(M_PI * get_element(temporary_lambda, i)) +
+              2.0 * M_PI * one_over_outer_radius_squared_ * square(x) *
+                  cos(M_PI * get_element(temporary_lambda, i)));
```
`temporary_lambda` is a double.  Remove `get_element`.

wthrowe reviewed (changes requested):
Just one minor thing.  Squash directly.

knelli2 commented:
@wthrowe Squashed
